server/Assignments_filler/sheetFiller.py

```diff
@@ -0,0 +1,237 @@
+    print (goodListOfNames)
+    print (listOfComments)
+
+    logging.info("Filling List...")
+
+    for comment in commentList:
+        cid = comment[0]
+        uid = comment[1]
+        subject = comment[2]
+        
+        #get user's attendence
+        query = ("SELECT `field_attendence_value` FROM drupal.field_data_field_attendence where entity_id=%s" % (cid))
+        attendence = get_from_db(query)[0][0]
+        
+        #ignore all comments not marked "yes"
+        if attendence != "yes":
+            continue
+        
+        #get user display name
+        query = ("SELECT `realname` FROM drupal.realname where uid=%s" % (uid))
+        name = get_from_db(query)[0][0]
+        
+        #check user's rank
+        query = ("SELECT `rid` FROM drupal.users_roles where uid=%s" % (uid))
+        rank = get_from_db(query)[0][0]
+        
+        #check for existing comment
+        #readRange = TEAM_ASSINGMENT_TAB + 'D' + str(row)
+        #localComment = readSheet(service, readRange)
+        localComment = ""
+
+        #if new mission, delete comment
+        #if name changed spot, get the old comment   
+        if nameChanged:
+            pass
+        elif name in goodListOfNames:
+            print (name)
+            index = goodListOfNames.index(name)
+            if index < (len(listOfComments)):
+                if listOfComments[index] == []:
+                    localComment = ""
+                else:
+                    localComment = listOfComments[index][0]
+                print (localComment)
+            else:
+                print ("no comment recorded for this name")
+                #pass
+
+        #if needed, add "recruit"
+        if rank == 4:
+            if localComment == "":
+                localComment = "Recruit"
+                print ("empty comment")
+            elif "Recruit" in localComment:
+                print ("comment contains recruit")
+                #pass
+            else:
+                print ("Recruit added to comment")
+                localComment = "Recruit, " + localComment
+            
+        #if no roles were filled, use "Rifleman"
+        if subject == "(No subject)":
+            subject = "Rifleman"
+
+        data = [name, subject, localComment]
+        writeRange = TEAM_ASSINGMENT_TAB + 'B' + str(row) + ":D" + str(row)
+        writeSheet(service, data, writeRange, 3)
+        row = row + 1
+    
+    #delete leftovers
+    logging.info("Deleting leftovers...")
+    data = ["", "", ""]
+    checkRange = TEAM_ASSINGMENT_TAB + 'B' + str(row) + ":D" + str(row)
+    NextCell = readSheet(service, checkRange, 3)
+    while NextCell != "":
+        writeSheet(service, data, checkRange, 3)
+        row = row + 1
+        checkRange = TEAM_ASSINGMENT_TAB + 'B' + str(row) + ":D" + str(row)
+        NextCell = readSheet(service, checkRange, 3)
+    
+    logging.info("Successfully filled!")
+    
+    checker(service, contractName, nid, numberOfComments)
```

Comment:
Checker calls Filler, Filler calls checker = cyclic dependency.

Avoid that, Checker can call Filler and that's it. It simply continues checking after that.